roofit/roofitcore/inc/RooAbsCollection.h

```diff
@@ -124,6 +124,7 @@ class RooAbsCollection : public TObject, public RooPrintable {
   RooAbsCollection* selectCommon(const RooAbsCollection& refColl) const ;
   RooAbsCollection* selectByName(const char* nameList, Bool_t verbose=kFALSE) const ;
   Bool_t equals(const RooAbsCollection& otherColl) const ; 
+  bool hasSameLayout(const RooAbsCollection& other) const;
```

Comment:
The function name `hasSameLayout` doesn't really tell me what the function actually does. Maybe there is a more self-explanatory name? Unfortunately I also can't think of another name that is reasonably short, so if you also can't think of something else spontaneously just leave it as it is.

Also, Nothing specific to the RooAbsCollection is used from `other`, I think it could be `std::span<RooAbsReal const*> other`.

Reply:
It was meant to check if you can use the `fast` argument of those functions in RooDataHist & co that either compare collections slowly or do the fast loop not checking the names.
`hasSameLayout` derives from
https://github.com/root-project/root/blob/17c1c1f455ff65e3db85bafb03d6dc18d0f8299a/roofit/roofitcore/src/RooDataHist.cxx#L961-L964

In this context, it's also a bit cumbersome to convert the collection into a span first.

If that gave you some ideas for a better name, I'll change it.

Reply:
I found `canUseFastCoordinateCheck` worse. 🙂

Reply:
I see, in the context of where it's used the name makes sense then. I'm fine with `hasSameLayout` then!

Reply:
This check is indeed interesting for some things that I tweaked (RooHistPdf). But in the case of RooHistPdf, the arg set had the same layout as the data hist layout by construction, so no check was necessary.

Reply:
@guitargeek I noticed something important:
You *must not* use `std::span` in ROOT's interfaces, because it's c++20. That's the reason why `RooSpan` exists.